### PR TITLE
[ISSUE-364][FEATURE] Worker should reject ReserveSlots request when shutting down

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
As designed, worker should reject new ReserveSlots request when shutting down

### Related issues.
https://github.com/alibaba/RemoteShuffleService/issues/364

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala [hunk 1]
```diff
@@ -76,7 +78,7 @@ private[deploy] class Controller(
           s"slave partitions: ${slaveLocations.asScala.map(_.getUniqueId).mkString(",")}.")
         handleReserveSlots(context, applicationId, shuffleId, masterLocations,
           slaveLocations, splitThreshold, splitMode, partitionType)
-        logDebug(s"ReserveSlots for $shuffleKey succeed.")
+        logDebug(s"ReserveSlots for $shuffleKey finished.")
```
Comment (AngersZhuuuu) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala:81]: Here not all case is succeed, should change to finished?

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala [hunk 2]
```diff
@@ -111,6 +113,13 @@ private[deploy] class Controller(
       splitMode: PartitionSplitMode,
       partitionType: PartitionType): Unit = {
     val shuffleKey = Utils.makeShuffleKey(applicationId, shuffleId)
+    if (shutdown.get()) {
+      val msg = "Current worker is shutting down!"
+      logError(s"[handleReserveSlots] $msg")
+      context.reply(ReserveSlotsResponse(StatusCode.ReserveSlotFailed, msg))
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala:119]: Shall we add a new status code like WorkerShuttingDown to be more specific?

Comment (AngersZhuuuu) [reply server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala:119]: > Shall we add a new status code like WorkerShuttingDown to be more specific?

Prefer to do this in a refactor pr for all such cases as we have discussed before

**Comment by AngersZhuuuu** [comment]
gentle ping

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**